src/VisualStudio/Xaml/Impl/Implementation/XamlOleCommandTarget.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.CodeAnalysis.Editor;
+using Microsoft.CodeAnalysis.Editor.Shared.Extensions;
+using Microsoft.CodeAnalysis.Editor.Xaml;
+using Microsoft.VisualStudio.Editor;
+using Microsoft.VisualStudio.LanguageServices.Implementation;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+
+namespace Microsoft.VisualStudio.LanguageServices.Xaml
+{
+    /// <summary>
+    /// This command target routes commands in .xaml files.
+    /// </summary>
+    internal sealed class XamlOleCommandTarget : AbstractOleCommandTarget
```

Comment:
Languages have a way to show/hide commands programmatically, we've just been historically lazy where a few command handlers just say "always visible" because there wasn't anybody else ever wanting to say anything else. @mgoertz-msft if you find any silliness on our side please just send the PRs and we'll get them merged.